WooCommerce/Classes/ViewRelated/Products/Edit Product/Product Settings/ProductSettingsSections.swift

```diff
@@ -20,8 +20,18 @@ enum ProductSettingsSections {
 
         let rows: [ProductSettingsRowMediator]
 
-        init(_ settings: ProductSettings) {
-            rows = [ProductSettingsRows.Status(settings), ProductSettingsRows.Visibility(settings), ProductSettingsRows.CatalogVisibility(settings)]
+        init(_ settings: ProductSettings, productType: ProductType) {
+            if ServiceLocator.featureFlagService.isFeatureFlagEnabled(.editProductsRelease3) && productType == .simple {
```

Comment:
I believe we should add a unit test for these configurations as these can probably get complicated, hard-to-track, and easy to break in the future. 

```swift
/// Test cases for `ProductSettingsSectinos.PublishSettings`.
///
final class ProductSettingsSections_PublishSettingsTests: XCTestCase {

    func testGivenANonSimpleProductThenItDoesNotShowTheVirtualProductOption() {
        // Given
        let settings = ProductSettings(status: .draft,
                                       featured: false,
                                       password: nil,
                                       catalogVisibility: .catalog,
                                       virtual: false,
                                       reviewsAllowed: false,
                                       slug: "",
                                       purchaseNote: nil,
                                       menuOrder: 0)
        let productType = ProductType.grouped

        // When
        let section = ProductSettingsSections.PublishSettings(settings, productType: productType)

        // Then
        XCTAssertNil(section.rows.first(where: {
            $0 is ProductSettingsRows.VirtualProduct
        }))
    }

    func testGivenASimpleProductThenItShowsTheVirtualProductOption() {
        // Given
        let settings = ProductSettings(status: .draft,
                                       featured: false,
                                       password: nil,
                                       catalogVisibility: .catalog,
                                       virtual: false,
                                       reviewsAllowed: false,
                                       slug: "",
                                       purchaseNote: nil,
                                       menuOrder: 0)
        let productType = ProductType.simple

        // When
        let section = ProductSettingsSections.PublishSettings(settings, productType: productType)

        // Then
        XCTAssertNotNil(section.rows.first(where: {
            $0 is ProductSettingsRows.VirtualProduct
        }))
    }
}

```

Reply:
Good idea, thanks for the suggestion @shiki! Implemented here 7695f7cf7fc500dc37ddc65827b1345ff8b10f4a